### PR TITLE
stm32/can: add option to enable deep-sleep per device

### PR DESCRIPTION
**Pull request by vincent-d**
### Contribution description

Deep-sleep was based on using rx pin as external interrupt to be able to
wake up from stop mode. If rx pin cannot be used as interrupt or user
does not need to wake up from stop from the CAN, an option is now
present. If en_deep_sleep_wake_up is set to false, setting the device to
sleep simply unblocks stop mode. Otherwise the behavior is unchanged.

### CODE REVIEW

File: cpu/stm32/include/candev_stm32.h [hunk 1]
```diff
@@ -100,6 +100,7 @@ typedef struct {
 #ifndef CPU_FAM_STM32F1
     gpio_af_t af;               /**< Alternate pin function to use */
 #endif
+    bool en_deep_sleep_wake_up; /**< Enable deep-sleep wake-up interrupt */
```
Comment (benpicco) [inline cpu/stm32/include/candev_stm32.h:103]: Why not extend the bit field below? (line 130)

Comment (vincent-d) [reply cpu/stm32/include/candev_stm32.h:103]: The bitfield was using `uint8_t` as the base type and there were already eight bits in it. Also, the bits in the bit field correspond to a setting in the config register.

**Comment by aabadie** [comment]
How can we test this ? I don't find any board that provides a default can configuration. The `tests/candev` application is also missing non native initialization code. This is out of the scope of this PR but would help if someone wants to try CAN on STM32.

**Comment by vincent-d** [comment]
@aabadie `tests/candev` is improved in #14909 so it can be used more easily with non-native devices. `tests/conn_can` might also be a good candidate for testing.

Normally, the nucleo-f413zh provides a CAN configuration, but I must admit this is not straightforward, as a default `can_params.h` is shared for all stm32 CPUs. You would also need a transceiver.

**Comment by aabadie** [comment]
So I guess the default can_params should be updated as well:
https://github.com/RIOT-OS/RIOT/blob/c6fe777388522f490cf50400d064ebc85d2defdd/cpu/stm32/include/can_params.h#L30-L31

>  the nucleo-f413zh provides a CAN configuration

There's nothing related to CAN at this level.

**Comment by fjmolinas** [comment]
> So I guess the default can_params should be updated as well:
> 
> https://github.com/RIOT-OS/RIOT/blob/c6fe777388522f490cf50400d064ebc85d2defdd/cpu/stm32/include/can_params.h#L30-L31
> 
> > the nucleo-f413zh provides a CAN configuration
> 
> There's nothing related to CAN at this level.

@vincent-d this one looks good, do you mind adding the default config as well?

**Comment by vincent-d** [comment]
Added `en_deep_sleep_wake_up`  in default can configs and rebased.

**Comment by fjmolinas** [comment]
All good, mind if I ask was what the use case for this is?

**Review by fjmolinas — APPROVED** [review]
ACK.

**Pull request merged**